Depl-217627 scmdb should fail if invalid objects are present after script execution

ov-automation opened the pull request:
[Depl-217627]( https://trackor.onevizion.com/form/ConfigAppForm.do?id=100106684746&ttid=10009161) scmdb should fail if invalid objects are present after script execution<br><br>[Depl-217627-107522](https://trackor.onevizion.com/form/ConfigAppForm.do?id=100107761816&ttid=100091445) dnesmelov (Dmitry Nesmelov)

MobyDrake committed:
Depl-217627 scmdb should fail if invalid objects are present after script execution
Depl-217627-99380 [DEV] scmdb should fail if invalid objects are present after script execution

Simplycissimys committed:
Merge pull request #70 from MobyDrake/Depl-217627

Depl-217627 scmdb should fail if invalid objects are present after script execution

Simplycissimys commented on src/main/java/com/onevizion/scmdb/DbManager.java:
```diff
@@ -79,15 +80,17 @@ private void executeNewScripts() {
             newCommitScripts.forEach(script -> logger.info(script.getName()));
-            newCommitScripts.forEach(script -> {
-                int exitCode = scriptExecutor.execute(script);
+            for (Iterator<SqlScript> iterator = newCommitScripts.iterator(); iterator.hasNext(); ) {
```
replace with `while`
```
Iterator<SqlScript> iterator = newCommitScripts.iterator();
while (iterator.hasNext()) {
```

MobyDrake replied:
reverted changes

Simplycissimys commented on src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java:
```diff
@@ -57,13 +55,13 @@ public class SqlScriptExecutor {
     private DataSource pkgDataSource;
 
-    public int execute(SqlScript script) {
+    public int execute(SqlScript script, boolean throwInvalidObject) {
```
to simplify code I propose to remove `throwInvalidObject` from `SqlScriptExecutor` and just call separate script after all executed scripts which will check invalid objects and throw exception if required

MobyDrake replied:
put the check in a separate script and method.

MobyDrake committed:
Depl-217627 scmdb should fail if invalid objects are present after script execution
Depl-217627-107522 [CR] scmdb should fail if invalid objects are present after script execution

MobyDrake committed:
Depl-217627 scmdb should fail if invalid objects are present after script execution
Depl-217627-107522 [CR] scmdb should fail if invalid objects are present after script execution

Simplycissimys committed:
Merge pull request #95 from MobyDrake/Depl-217627

Depl-217627 scmdb should fail if invalid objects are present after script execution

Simplycissimys commented on src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java:
```diff
@@ -158,4 +160,36 @@ private Connection getConnection(SchemaType schemaType, String schemaName) {
         }
     }
 
+    public void checkInvalidObjectAndThrow() {
```
rename to `checkInvalidObjectOrExit()`

MobyDrake replied:
done.

Simplycissimys commented on src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java:
```diff
@@ -158,4 +160,36 @@ private Connection getConnection(SchemaType schemaType, String schemaName) {
         }
     }
 
+    public void checkInvalidObjectAndThrow() {
```
in https://github.com/Onevizion/scmdb/pull/88 was implemented common method to execute scripts, wait when it if will be merge to the master ans use it

MobyDrake replied:
Updated this branch. Because of custom exception and exception message get from contex, i changed signature of methods  `execute()` - now they return context.

MobyDrake committed:
Merge remote-tracking branch 'upsource/master' into Depl-217627

MobyDrake committed:
Merge branch 'master' into Depl-217627

# Conflicts:
#	src/main/java/com/onevizion/scmdb/DbManager.java
#	src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java

MobyDrake committed:
Depl-217627 scmdb should fail if invalid objects are present after script execution
Depl-217627-107522 [CR] scmdb should fail if invalid objects are present after script execution

MobyDrake committed:
Depl-217627 scmdb should fail if invalid objects are present after script execution
Depl-217627-107522 [CR] scmdb should fail if invalid objects are present after script execution

Simplycissimys committed:
Merge pull request #106 from MobyDrake/Depl-217627

Depl 217627

Simplycissimys reviewed (changes requested):
fix conflict

Commit:
Merge remote-tracking branch 'source/master' into Depl-217627

# Conflicts:
#	src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java

Commit:
Depl-217627 scmdb should fail if invalid objects are present after script execution
Depl-217627-107522 [CR] scmdb should fail if invalid objects are present after script execution

Simplycissimys committed:
Merge pull request #120 from MobyDrake/Depl-217627

Depl-217627

Simplycissimys commented on src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java:
```diff
@@ -186,4 +202,25 @@ private Connection getConnection(SchemaType schemaType, String schemaName) {
+        }
+    }
+
+    private int getExitCode(ScriptRunnerContext context) {
```
working with `exitCode` is spread across classes. I propose to refactor `SqlScriptExecutor`. It should return `void` and throw exceptions if something fails, the exit code should be calculated inside `Scmdb` class

MobyDrake replied:
Done

MobyDrake replied:
Note: now catch() use `System.err.print` because `ColorLogger` reset logger configurate. We must create more powerfull logger for `Scmdb.class`

Commit:
Depl-217627 scmdb should fail if invalid objects are present after script execution
Depl-217627-107522 [CR] scmdb should fail if invalid objects are present after script execution

Simplycissimys committed:
Merge pull request #121 from MobyDrake/Depl-217627

Depl-217627 scmdb should fail if invalid objects are present after script execution
